### PR TITLE
Let Flutter tools get material icons from pub package instead of special logic

### PR DESCRIPTION
**Pull request by xster**
- Removes logic to embed fonts based on the `uses-material-design:` flag in pubspec.yaml
- Stop caching the material fonts in Flutter tools artifacts
- Update material Icons to refer to glyphs from the `material_icons_font` package
- Update the font update script
- Add a project pubspec upgrade warning message for existing projects
- Tests

After submit:
- [ ] Update https://github.com/flutter/flutter/wiki/Updating-Material-Design-Fonts 
- [ ] Send announcement
- [ ] Update [all usages in the Flutter org](https://github.com/search?p=3&q=org%3Aflutter+uses-material-design&type=Code)
- [ ] Rename the cupertino_icons package to cupertino_icons_font for consistency (because material_icons is already taken on pub)

_API BREAK_
Will require manual upgrade from [~1000 project pubspec.yaml on GitHub](https://github.com/search?utf8=%E2%9C%93&q=uses-material-design&type=Code)

## Upgrade instructions
For existing projects, go to pubspec.yaml and change
```
name: [...]
dependencies:
  [...]
flutter: 
  uses-material-design: true
  [...]
```
into
```
name: [...]
dependencies:
  [...]
  material_icons_font: ^0.1.1
flutter:
  [...]
```
to continue to use the icons assets referred to by the Material [Icons class](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/icons.dart)

### CODE REVIEW

File: packages/flutter_tools/lib/src/asset.dart [hunk 1]
```diff
@@ -22,9 +19,25 @@ class AssetBundle {
...
-  static const String _kFontSetMaterial = 'material';
   static const String _kLICENSE = 'LICENSE';
 
+  static const String _materialIconsFontUpgradeInstructions = '''
+═══════════════════════════════════════════════════════════════════════════════════
+'uses-material-design' in pubspec.yaml is deprecated
+
+The flag is now replaced by a generic Dart package to provide the Material icons asset.
+
+Replace
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/asset.dart]: /cc @bwilkerson, @pq - an example of what would make a good lint for pubspec files

Comment (pq) [reply packages/flutter_tools/lib/src/asset.dart]: Interesting!

If someone could capture the advice in a sentence (or more), I'll open a lint request.  Thanks!

**Comment by eseidelGoogle** [comment]
Is there a way we could provide a transition period, or support the old format through some tool hacks for a while?  Breaking all Flutter apps is certainly possible, just becoming more and more costly every day.

**Comment by xster** [comment]
I agree it's a bad impression to give to users of thousands of projects to upgrade their source manually. 

I think it may be the last straw to get us to bite the bullet on https://github.com/flutter/flutter/issues/12573. 

The other alternative that's also long-term-ish is to have a specified queue of font family fallbacks that could solve this too. 

Unfortunately I don't have good ideas for a very quick hack to solve this. Since the previous way integrated deep during the flx building and the new way pub gets before the flutter tools dart code runs

**Comment by xster** [comment]
Gonna bite the bullet and build the #12573 mechanism before submitting this. We run into this frequently enough that it's going to benefit a number of things.

**Comment by Hixie** [comment]
@xster Seems reasonable. Can we close this PR in the meantime? (Just to get it off the review list.)

**Pull request closed without merging**